### PR TITLE
ci(frontend): add script that checks if there are ERC20 duplicates

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

Created script to avoid having duplicates between the ERC20 tokens in folder `env`. To run it we require [vite-node](https://www.npmjs.com/package/vite-node) (thanks to @peterpeterparker for the suggestion).

# Tests

Manual test:

1. Run script.
2. No error given (CORRECT).
3. Copy an ERC20 address from `networks.icrc.env.ts` and paste it in one of the tokens in `env/tokens-erc20`.
4. Run script.
5. Raises error (CORRECT).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,8 @@
 		"e2e:dev": "NODE_ENV=development playwright test",
 		"e2e:ci": "playwright test --reporter=html",
 		"e2e:snapshots": "npx playwright test --update-snapshots --reporter=list",
-		"e2e:report": "npx playwright show-report"
+		"e2e:report": "npx playwright show-report",
+		"erc20:check": "npx vite-node scripts/check.tokens.erc20.ts"
```
Comment (peterpeterparker) [inline package.json:42]: Shouldn't it be `npm run vite-node` or just `vite-node` given that the dependencies is installed?

File: scripts/check.tokens.erc20.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import type { Erc20Contract } from '$eth/types/erc20';
+import {
+	ERC20_CONTRACTS_PRODUCTION,
+	ERC20_CONTRACTS_SEPOLIA,
+	ERC20_TWIN_TOKENS_MAINNET,
+	ERC20_TWIN_TOKENS_SEPOLIA
+} from '../src/frontend/src/env/tokens.erc20.env';
+
+const getAddresses = (contracts: Erc20Contract[]) =>
```
Comment (peterpeterparker) [inline scripts/check.tokens.erc20.ts:9]: NTH: Given that's a TS script, can you express the return types of the function?

For example:

```
const getAddresses = (...): boolean => 
```

File: scripts/check.tokens.erc20.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	contracts.map((contract) => contract.address.toLowerCase());
+
+const productionAddresses = getAddresses(ERC20_CONTRACTS_PRODUCTION);
+const twinTokensMainnetAddresses = getAddresses(ERC20_TWIN_TOKENS_MAINNET);
+
+const sepoliaAddresses = getAddresses(ERC20_CONTRACTS_SEPOLIA);
+const twinTokensSepoliaAddresses = getAddresses(ERC20_TWIN_TOKENS_SEPOLIA);
+
+const checkDuplicates = (addresses1: string[], addresses2: string[]) =>
```
Comment (peterpeterparker) [inline scripts/check.tokens.erc20.ts:18]: NTH: This function does not really "check", it "filter" the addresses

File: scripts/check.tokens.erc20.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		console.error(
+			'Error: Duplicate ERC20 token addresses found between ERC20_CONTRACTS_SEPOLIA and ERC20_TWIN_TOKENS_SEPOLIA:',
+			sepoliaDuplicates
+		);
+	}
+
+	process.exit(1);
+} else {
+	console.log('No ERC20 token duplicates found. All addresses are unique.');
+	process.exit(0);
```
Comment (peterpeterparker) [inline scripts/check.tokens.erc20.ts:42]: You don't need this line

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

File: scripts/check.tokens.erc20.ts (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply scripts/check.tokens.erc20.ts:9]: done

File: scripts/check.tokens.erc20.ts (hunk 3 above)
Comment (AntonioVentilii-DFINITY) [reply scripts/check.tokens.erc20.ts:18]: right! naming!

File: package.json (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply package.json:42]: you are right! I forgot to change it there...

**Pull request merged**